share/doc/homebrew/Node-for-Formula-Authors.md

```diff
@@ -0,0 +1,115 @@
+# Node for formula authors
+
+This document explains how to successfully use Node and npm in a node module based Homebrew formula.
+
+# Running `npm install`
+
+Homebrew provides two helper methods in a `Language::Node` module, `std_npm_install_args` and `local_npm_install_args`. They both set up the correct environment for npm and return arguments for `npm install` for their specific use cases. Please use them instead of invoking `npm install` explicitly. The syntax for a standard node module installation is:
+
+```ruby
+system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+```
+
+where `libexec` is the destination prefix (usually the `libexec` variable).
+
+# Download URL
+
+If the node module is also available on the npm registry, we prefer npm hosted release tarballs over github (or elsewhere) hosted source tarballs. The advantages of these tarballs are that they doesn't include the files from the `.npmignore` (such as tests) resulting in a smaller download size and that a possibly transpilation step is already done (e.g. no need to compile Coffee Script files as a build step).
+
+The npm registry URLs have usually the format of:
+
+```
+https://registry.npmjs.org/<name>/-/<name>-<version>.tgz
+```
+
+Alternatively you could curl the JSON at `https://registry.npmjs.org/<name>` and look for the value of `versions[<version>].dist.tarball` for the correct tarball URL.
+
+# Dependencies
+
+Node modules, which are compatible with the latest Node version should declare a dependencies on the `node` formula.
+
+```ruby
+depends_on "node"
+```
+
+If your formula requires to be executed with an older Node version you must vendor this older Node version as done in the [`kibana` formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/kibana.rb).
```

Comment:
Can you peg the formulae references to a specific commit? That way if `kibana` or anything else stops using `node` in future the reference remains likely to be valid still.